api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryService.java

```diff
@@ -24,17 +24,41 @@
 public interface ReportingQueryService {
   long getQueryBatchSize();
 
-  List<ReportingDataset> getDatasets();
+  List<ReportingDataset> getDatasetBatch(long limit, long offset);
```

Comment:
always get a batch of these data types, instead of the whole table like we would sometimes do.

TODO: there's a lot of boilerplate in the reporting system.  I have some ideas to streamline in a later PR.

Reply:
https://github.com/all-of-us/workbench/pull/9096